sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -149,11 +149,8 @@ statement
     | ALTER (TABLE | VIEW) multipartIdentifier
         UNSET TBLPROPERTIES (IF EXISTS)? tablePropertyList             #unsetTableProperties
     | ALTER TABLE multipartIdentifier
-        (ALTER | CHANGE) COLUMN? qualifiedName
+        (ALTER | CHANGE) COLUMN? multipartIdentifier
```

Comment:
Previously we use `qualifiedName: identifier ('.' identifier)*` to capture column name. 

This conflicts a test in ErrorParserSuite that `test-col` is not allowed in `ALTER TABLE t CHANGE COLUMN test-col TYPE BIGINT`.

The column name should be multiple errorCapturingIdentifier. So I changed it to multipartIdentifier:
```
multipartIdentifier
    : parts+=errorCapturingIdentifier ('.' parts+=errorCapturingIdentifier)*
    ;
```

@cloud-fan

Reply:
ok. will make a followup later.